src/components/servo/client.hpp

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include <grpc/support/log.h>
+#include <grpcpp/grpcpp.h>
+
+#include "servo.hpp"
+#include "component/servo/v1/servo.grpc.pb.h"
+
+using grpc::Channel;
+using grpc::ClientContext;
+using viam::component::servo::v1::ServoService;
+
+class ServoClient: public Servo {
+    public:
+        explicit ServoClient(std::string name_, std::shared_ptr<Channel> channel);
+        static std::shared_ptr<ServoClient> from_robot(std::shared_ptr<RobotClient> robot, std::string name);
+        void move(int angle);
```

Comment:
Is `ServoClient::move` intended to be an override of `Servo::move`? As things are currently set up, it is not, because `Servo::move` is non-virtual.

Reply:
I think yes, and I remember making this non-virtual to solve a different issue.

So the way the other SDKs work is that when we call `from_robot`, we actually want to call it on the base component itself, not the client (in this case, on `Servo`, not `ServoClient`). This is because we want the robot to return anything that implements that component, so in this case, anything that is `Servo`-like. 

Is that something that's possible in CPP? Could I make `Servo` abstract, and then have `from_robot` return `Servo`, but in reality it is a `ServoClient` (since `ServoClient` implements `Servo`)?

Reply:
Yes, definitely possible, and more or less the canonical example of where you use virtual methods / abstract classes. Something like this:

```
class Servo: public ComponentBase {
    public:
        static ComponentType type;
        virtual void move(int angle) = 0;
        virtual int get_position() = 0;
        virtual bool is_moving() = 0;
        ...
};
```

Then 

```
class ServoClient: public Servo {
    public:
        ...
        void move(int angle) override {...}
        int get_position() override { ... }
        bool is_moving() override { ... }
};
```

The key things here are declaring the functions in `Servo` as `virtual`, and then making those methods as pure virtual with `=0`. The use of `override` in the deriving class is not required, but it is a good practice.

But before doing this, can you provide some background on why there is a need for both `Servo` and `ServoClient`? Are there going to be things that are`Servo`s but aren't `ServoClients`?